docs/scripts/object_orientated_programming_poc/oop_poc_user_and_permission.py

```diff
@@ -0,0 +1,97 @@
+"""The purpose of this script is to demonstrate how to use the OOP interface for
+UserProfile and permissions on some entities.
+The following actions are shown in this script:
+1. Get a UserProfile by name and id
+2. Check if a UserProfile is certified
+3. Set permissions on a File
+4. Get permissions on a File
+"""
+import asyncio
+import os
+
+from synapseclient.models import (
+    File,
+    UserProfile,
+)
+import synapseclient
+
+PROJECT_ID = "syn52948289"
+DPE_TEAM_SERVICE_ACCOUNT_NAME = "synapse-service-dpe-team"
+DPE_TEAM_SERVICE_ACCOUNT_ID = 3485485
+
+syn = synapseclient.Synapse(debug=True)
+syn.login()
+
+
+def create_random_file(
+    path: str,
+) -> None:
+    """Create a random file with random data.
+
+    :param path: The path to create the file at.
+    """
+    with open(path, "wb") as f:
+        f.write(os.urandom(1))
+
+
+async def user_profile():
+    dpe_team_profile = await UserProfile(username=DPE_TEAM_SERVICE_ACCOUNT_NAME).get()
+    print(f"Getting account by name: {dpe_team_profile}")
+
+    dpe_team_profile = await UserProfile(id=DPE_TEAM_SERVICE_ACCOUNT_ID).get()
+    print(f"Getting account by id: {dpe_team_profile}")
+
+    dpe_team_profile = await UserProfile.from_username(
+        username=DPE_TEAM_SERVICE_ACCOUNT_NAME
+    )
+    print(f"Getting account by name: {dpe_team_profile}")
+
+    dpe_team_profile = await UserProfile.from_id(user_id=DPE_TEAM_SERVICE_ACCOUNT_ID)
+    print(f"Getting account by id: {dpe_team_profile}")
+
+    print(
+        f"is certified - By ID?: {await UserProfile(id=DPE_TEAM_SERVICE_ACCOUNT_ID).is_certified()}"
+    )
+    print(
+        f"is certified - By name?: {await UserProfile(username=DPE_TEAM_SERVICE_ACCOUNT_NAME).is_certified()}"
+    )
+
+
+async def set_permissions():
+    name_of_file = "my_file_with_read_permissions_for_service_account.txt"
+    path_to_file = os.path.join(os.path.expanduser("~/temp"), name_of_file)
+    create_random_file(path_to_file)
+
+    file_with_read = await File(
```

Comment:
@BryanFauble I am running into permissions issues here
```
synapseclient.core.exceptions.SynapseHTTPError: 403 Client Error: 
Lack of READ permission on the requested entity.
```

Reply:
My mistake - During testing I removed the DPE team from sharing settings on the File in order to verify that only the user permissions were being applied (Instead of BOTH the user and team permissions). 

What this means is because the DPE team service account is a member of the DPE team when you create the local sharing settings and print out the effective ACL it's going to be the union of both ACL sets.

Instead of using the DPE team service account I swapped this to use my other prod testing account. You should be able to run the script now @BWMac